### PR TITLE
RTSP recording application

### PR DESCRIPTION
**Pull request by FFFFFaraway**
**What this PR does / why we need it**:
RTSP recording application running in the kubernetes allow user to connect multiple RTSP server concurrently, and record the RTSP stream into file like mp4.

**One page design**: https://github.com/Edgenesis/shifu/blob/main/docs/design/design-rtsp-record-application.md

1. Support the username and password secret provided by the user.
2. Support state persistence in the form of snapshot, and automatically restart all interrupted tasks.
3. Support the persistence of the stored video.

**Will this PR make the community happier**? 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**How is this PR tested**
- [x] unit test
- [x] e2e test
- [ ] other (please specify)

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cmd/rtspRecord/main.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/edgenesis/shifu/pkg/logger"
+	"github.com/edgenesis/shifu/pkg/rtspRecord"
+	"net/http"
+	"os"
+)
+
+var serverListenPort = os.Getenv("SERVER_LISTEN_PORT")
+
+const storePersistFilePath = "/data/mapStore"
+const videoPersistDirectory = "/data/video"
```
Comment (BtXin) [inline cmd/rtspRecord/main.go:13]: const ()

File: pkg/rtspRecord/map_store.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
+package rtspRecord
+
+import (
+	"encoding/gob"
+	"github.com/edgenesis/shifu/pkg/logger"
+	"os"
+	"sync"
+)
+
+type PersistMap struct {
```
Comment (BtXin) [inline pkg/rtspRecord/map_store.go:10]: Please add some comments on what does this structs do, and add some comments on the methods as well.

File: pkg/rtspRecord/rtsp_record.go [hunk 3]
```diff
@@ -0,0 +1,106 @@
+package rtspRecord
```
Comment (BtXin) [inline pkg/rtspRecord/rtsp_record.go:1]: Add some comments for this file as well.

File: pkg/rtspRecord/util.go [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	"encoding/json"
+	"fmt"
+	"github.com/edgenesis/shifu/pkg/deviceshifu/deviceshifubase"
+	"github.com/edgenesis/shifu/pkg/logger"
+	"github.com/edgenesis/shifu/pkg/telemetryservice/utils"
+	"io"
+	"net/http"
+)
+
+func trans[T Request](r *http.Request) (*T, error) {
```
Comment (BtXin) [inline pkg/rtspRecord/util.go:13]: Add some comments briefly describe what does this do.

**Comment by leepala** [comment]
@FFFFFaraway  Let's move those code into a new repo [github](https://github.com/Edgenesis/RTSPApplication)

**Pull request closed without merging**